packages/client/src/rpc/modules/eth.ts

```diff
@@ -200,6 +203,53 @@ const jsonRpcReceipt = async (
   blobGasPrice: blobGasPrice !== undefined ? bigIntToHex(blobGasPrice) : undefined,
 })
 
+const calculateRewards = async (
+  block: Block,
+  receiptsManager: ReceiptsManager,
+  priorityFeePercentiles: bigint[]
+) => {
+  const blockRewards: bigint[] = []
+  const txGasUsed: bigint[] = []
+  const baseFee = block.header.baseFeePerGas
+  const receipts = await receiptsManager.getReceipts(block.hash())
+
+  if (receipts.length > 0) {
+    txGasUsed.push(receipts[0].cumulativeBlockGasUsed)
+    receipts.shift()
+  }
+
+  for (const receipt of receipts) {
+    txGasUsed.push(receipt.cumulativeBlockGasUsed - txGasUsed[txGasUsed.length - 1])
```

Comment:
Shouldn't we subtract `receipts[n-1].cumulativeBlockGasUsed` here and not just the gas for the previous tx?

Reply:
Yes you are right :)

Reply:
We care about the tx gas used and not about the cumulative gas used (this is fixed).